pkg/distributor/distributor.go

```diff
@@ -786,6 +786,10 @@ func (d *Distributor) Push(ctx context.Context, req *cortexpb.WriteRequest) (*co
 		if sp := opentracing.SpanFromContext(ctx); sp != nil {
 			localCtx = opentracing.ContextWithSpan(localCtx, sp)
 		}
+		// Get any HTTP headers that are supposed to be added to logs and add to localCtx for later use
+		if headerMap := ctx.Value(util_log.HeaderMapContextKey); headerMap != nil {
+			localCtx = context.WithValue(localCtx, util_log.HeaderMapContextKey, headerMap)
```

Comment:
Maybe use the same pattern as open tracting. They do have helper getter and setter